spot_rl_experiments/spot_rl/utils/test_vrs_mps.py

```diff
@@ -0,0 +1,163 @@
+                    2,
+                    cv2.LINE_AA,
+                )
+            cv2.imshow(f"Stream - {name}", image)
+        cv2.waitKey(1)
+
+    def create_display_windows(self, stream_name_list: list):
+        for stream_name in stream_name_list:
+            cv2.namedWindow(f"Stream - {stream_name}", cv2.WINDOW_NORMAL)
+
+    def rectify_image(self, stream_name: str, image):
+        src_calib = self._src_calib_params_dict.get(stream_name)
+        dst_calib = self._dst_calib_params_dict.get(stream_name)
+        rectified_image = calibration.distort_by_calibration(
+            image, dst_calib, src_calib
+        )
+        return rectified_image
+
+    def stream_cameras(self, stream_names: list, should_rectify=True, detect_qr=False):
+        start_time = time.time()
+        self.create_display_windows(stream_name_list=stream_names)
+
+        # Setup rectification
+        device_calib = self.provider.get_device_calibration()
+        if should_rectify:
+            for stream_name in stream_names:
+                if stream_name not in self._src_calib_params_dict.keys():
+                    self._src_calib_params_dict[
+                        stream_name
+                    ] = device_calib.get_camera_calib(stream_name)
+                    self._dst_calib_params_dict[
+                        stream_name
+                    ] = calibration.get_linear_camera_calibration(
+                        512, 512, 280, stream_name
+                    )
+
+        # Setup April tag detection
+        if detect_qr:
+            for stream_name in stream_names:
+                if stream_name not in self._qr_pose_estimator_dict.keys():
+                    focal_lengths = self._dst_calib_params_dict.get(
+                        stream_name
+                    ).get_focal_lengths()
+                    principal_point = self._dst_calib_params_dict.get(
+                        stream_name
+                    ).get_principal_point()
+                    calib_dict = {
+                        "fx": focal_lengths[0].item(),
+                        "fy": focal_lengths[1].item(),
+                        "ppx": principal_point[0].item(),
+                        "ppy": principal_point[1].item(),
+                        "coeffs": [0.0, 0.0, 0.0, 0.0, 0.0],
+                    }
+
+                    # LETS ONLY DETECT DOCK-ID RIGHT NOW
+                    april_tag_pose_estimator = AprilTagPoseEstimator(
+                        camera_intrinsics=calib_dict
+                    )
+                    april_tag_pose_estimator.register_marker_ids([DOCK_ID])
+                    self._qr_pose_estimator_dict[stream_name] = april_tag_pose_estimator
+
+        # Assuming all frames are synced
+        stream_id = self.provider.get_stream_id_from_label(stream_names[0])
+        # num_frames = self.provider.get_num_data(stream_id)
+        # CUSTOM RANGE FOR VIDEO - TODO: REMOVE LATER
+        custom_range = range(1400, 1550)
```

Comment:
Get `start_idx` and `end_idx` as arguments (ints, None by default) and create custom_range that way. Don't delete this, I think this is a good utility.

Reply:
addressed